src/main/java/com/amazonaws/services/kinesis/leases/impl/LeaseManager.java

```diff
@@ -582,7 +583,9 @@ public boolean updateLease(T lease)
         UpdateItemRequest request = new UpdateItemRequest();
         request.setTableName(table);
         request.setKey(serializer.getDynamoHashKey(lease));
-        request.setExpected(serializer.getDynamoLeaseCounterExpectation(lease));
+        Map<String, ExpectedAttributeValue> expectations = serializer.getDynamoLeaseCounterExpectation(lease);
+        expectations.putAll(serializer.getDynamoLeaseCheckpointExpectation(lease));
+        request.setExpected(expectations);
```

Comment:
Won't this fail? I believe this is also used for leaseRenewal codepath, where we're heartbeating the lease counter. We don't necessarily expect `SHARD_END` in this case.

Reply:
I don't see this being used anywhere other than checkpoint code path. And we don't expect SHARD_END here. We expect it not to present when we write a non SHARD_END checkpoint.